### ISSUE TITLE
[Bug]: no images on the distribution plots

### ISSUE THREAD

**Issue by renhao2000** [jasp-stats/jasp-issues#1671]
### JASP Version

0.16.1 (Development)

### What analysis are you seeing the problem on?

descriptiveties

### What OS are you seeing the problem on?

win10

### Bug Description

after uploading the data, it shows the descriptive statistics but is blank in the distribution plots

### Expected Behaviour

plz help me to exhibit the plot

### Steps to Reproduce

1. menu
2. open the file
3. descriptives
..4. distribution plot

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if any.
- [X] I have included a JASP file that causes the crash/bug, if any.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by renhao2000** [comment]
![image](https://user-images.githubusercontent.com/102519767/160379874-a3137d0e-5766-4e99-b76c-1643227b0d5d.png)

**Comment by icekylin** [comment]
@renhao2000 Hi, 
First of all, Could you please attach a copy of your jasp data file here (you can compress it as a zip package and drag and drop to upload it in the comment area),so that we can try the process.
In addition, I noticed that there is a path named in Chinese string '栗' in your working directory, you can put this file in a non-Chinese path and try to see if there is still the same problem. And feedback here, thank you!

**Comment by renhao2000** [comment]
really appreciate for ur help, I just download old version, which does work, really appreciate it!

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Shun Wang ***@***.***>
发送时间: Monday, March 28, 2022 10:15:40 PM
收件人: jasp-stats/jasp-issues ***@***.***>
抄送: renhao2000 ***@***.***>; Mention ***@***.***>
主题: Re: [jasp-stats/jasp-issues] [Bug]: no images on the distribution plots (Issue #1671)


@renhao2000<https://github.com/renhao2000> Hi,
First of all, Could you please attach a copy of your jasp data file here (you can compress it as a zip package and drag and drop to upload it in the comment area),so that we can try the process.

In addition, I noticed that there is a path named in Chinese string '栗' in your working directory, you can put this file in a non-Chinese path and try to see if there is still the same problem. And feedback here, thank you!

―

**Comment by NYC0326** [comment]
I'm having the same error, could you help me out? Which version did you downloaded?

**Comment by renhao2000** [comment]
i found 0.15verson

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: NYC0326 ***@***.***>
发送时间: Tuesday, March 29, 2022 3:21:27 PM
收件人: jasp-stats/jasp-issues ***@***.***>
抄送: renhao2000 ***@***.***>; Mention ***@***.***>
主题: Re: [jasp-stats/jasp-issues] [Bug]: no images on the distribution plots (Issue #1671)


I'm having the same error, could you help me out? Which version did you downloaded?

―

**Comment by icekylin** [comment]
> I'm having the same error, could you help me out? Which version did you downloaded?

@NYC0326 
I got reports of this from several Chinese JASP users, but have not been able to reproduce the process. Can you provide a JASP file and log file containing this error (you can compress the .jasp file into a .zip file and upload it in this https://github.com/jasp-stats/jasp-issues/issues/1671 comment section, the log file you can find the path in the JASP preferences settings), and Provide the working path name of your JASP file.

I suspect it is because the path contains Unicode characters, which disappears when users switch to English paths. any way, we hope to get an example that can reproduce the problem.

Until then you can try an older version of JASP as @renhao2000  said.

**Comment by NYC0326** [comment]
> but have not been able to reproduce the process

I have deleted remaining files as I downgraded so I don't have log files.
I have put my csv files in an English paths in case that it is an unicode error, but it also didn't worked.

**Comment by icekylin** [comment]
@juliuspf sure, i'd love to look into this if someone give a JASP file to reproduce it.

**Comment by shun2wang** [comment]
Now I can reproduce and confirm that this has something to do with character handling of file paths.

1. If there is a user name composed of non-ASCII characters on Windows, the user name cannot be read correctly, resulting in R not being able to plot. The same problem occurs with R(4.1.x)/rstudio..
[jaspdata.jasp.zip](https://github.com/jasp-stats/jasp-issues/files/8582528/jaspdata.jasp.zip)

2. It is normal to change the system username or add a new one with English characters .

so,first i'll wait for a 0.16.3 build on R4.2, or if i guessing this might involve C++ code for the data reading part (I'm not familiar with C++), maybe @JorisGoosen can take a look?

**Comment by JorisGoosen** [comment]
@shun2wang I think you are probably right. Im still on vacation but I think @amirmasoudabdol might get the 4.2.0 R into JASP before he leaves and that would solve it. So hopefully by the time my vacation is over the problem will have solved itself ;)

**Comment by shun2wang** [comment]
Hi Joris @JorisGoosen , that https://github.com/jasp-stats/jasp-desktop/pull/4763 won't this PR be merged into 0.16.3 right? However, for the previous speculation, I think this may be related to the path of C++ processing `TempFiles` and `ShareMemory`, because when I reproduced the problem, I found that no cache file was generated in the corresponding path. Whether this has something to do with cache file IO between engine, data read and R, maybe libboost_nowide or other.
The following surveys I did for your reference:
1. For a Windows console, in Windows 10 (version 1903) and above systems, check and set the code page of the system, and run jasp.exe in the cmd window, the problem still exists.
2. Looking at the reproduce log, we can see that the path with Chinese character in the log is `C:/Users/杨测试/AppData/Local/JASP/temp/` 
 but the log it recorded in different encoding formats（you can change encoding in same log file and see `TempFiles::addShmemFileName ` and `Error when retrieving the data set:` in log file)
[JASP 2022-05-06 20_37_29 Desktop.log](https://github.com/jasp-stats/jasp-issues/files/8814279/JASP.2022-05-06.20_37_29.Desktop.log)
[JASP 2022-05-06 20_37_29 Engine 0.log](https://github.com/jasp-stats/jasp-issues/files/8814280/JASP.2022-05-06.20_37_29.Engine.0.log)
And when I look at the corresponding path, the folder is empty.

In summary we are getting closer, can you take a look at it? thanks

**Comment by JorisGoosen** [comment]
hey @shun2wang nice detective work!

Too bad that PR won't be merged for 0.16.3 because it would change the buildsystem some more and we are having trouble normally building with the current code. But it will be in in the next version.

A few remarks though:
`TempFiles::addShmemFileName` maybe once had a function but it always gets the wrong path as it turns out (so you couldnt find those files because they were never there, nor on any system...). Which is also the reason I removed that function in https://github.com/jasp-stats/jasp-desktop/pull/4726 

I honestly don't want to spend too much time on this issue until we get https://github.com/jasp-stats/jasp-desktop/pull/4763 merged because that should solve all the problems we are seeing automatically anyway.
These problems happen because of certain encoding flags that R sets up automatically and for any strings we pass we have to be very careful to convert them to that encoding or flag it as being utf8. So it seems this goes wrong somewhere, maybe even inside an R package where it ignores the encoding in some C-code or something. I don't know.

But Ive looked at enough of these bugs and problems to know it can take a lot of time to figure out where what goes wrong.

I think it is better to invest that time in getting JASP working with R 4.2.0.
